src/krb5_asrep_common_plug.c

```diff
@@ -47,15 +46,14 @@ char *krb5_asrep_split(char *ciphertext, int index, struct fmt_main *self)
 			p = strchr(ciphertext + FORMAT_TAG_LEN + ETYPE_TAG_LEN + 1, '$') + 1;
 			for (i = 0; i < p - ciphertext; i++)
 				ptr[i] = ARCH_INDEX(ciphertext[i]);
```

Comment:
The above loop (not touched by the changes here yet) looks like a weird way to do `memcpy` - so perhaps simplify it? You'd also remove the variable `i` entirely, and replace your two `+=` lines below with assignments of the correct pointer values.